packages/@aws-cdk/aws-codepipeline/lib/stage.ts

```diff
@@ -146,6 +146,14 @@ export class Stage extends cdk.Construct implements actions.IStage {
     }
   }
 
+  public _generateOutputArtifactName(action: actions.Action): string {
+    return (this.pipeline as any)._generateOutputArtifactName(this, action);
```

Comment:
Honestly I don't like having to do this. I'd rather actually make the methods public & mark them s internal-use-only, which is already semi-obvious from their `_`-prepended names.

Reply:
Does it make much difference though?

Reply:
Main difference is the compiler stops trying to help the moment you've done `as any`, so you might miss breaking changes if you rename stuff & there is no test coverage through there.